### DISCUSSION TITLE
JavaScript actions support for node18 or node20

### DISCUSSION THREAD

**Discussion by MikeMcC399** [community/community#53217] [Actions] [answered]
### Select Topic Area

Product Feedback

### Body

Only `node16` is available for JavaScript actions according to [`runs.using` for JavaScript actions](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsusing-for-javascript-actions).

This is not enough!

Some JavaScript actions like [cycjimmy/semantic-release-action](https://github.com/cycjimmy/semantic-release-action) are blocked from using later dependencies due to this.

See [Semantic version](https://github.com/cycjimmy/semantic-release-action#semantic_version) which says:

"*Note: [Version 20.0.0](https://github.com/semantic-release/semantic-release/releases/tag/v20.0.0) of [semantic-release](https://github.com/semantic-release/semantic-release/) and later is currently incompatible with `cycjimmy/semantic-release-action`, since it requires Node.js 18. GitHub does not provide this environment yet for [JavaScript actions](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runs-for-javascript-actions).*"

Please add `node18` or `node20` to the available versions for [`runs.using` for JavaScript actions](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsusing-for-javascript-actions).

According to https://github.com/nodejs/release#release-schedule:
Node.js 16 end-of-life is scheduled for Sep 11, 2023
Node.js 18 end-of-life is scheduled for Apr 30, 2025
Node.js 20 starts Active LTS on Oct 24, 2023 and its end-of-life is scheduled for Apr 30, 2026

**Comment by MikeMcC399** [comment]
Can we expect some feedback from product management on this important topic?

https://github.com/actions/runner/issues/new/choose shows:

![image](https://user-images.githubusercontent.com/66998419/233597102-3c35440a-16dd-4a1f-af59-f3125bc62906.png)

**Comment by MikeMcC399** [comment]
Node.js `16`, which is currently the only supported version for GitHub JavaScript actions, is planned to enter End-of-Life status on Sep 11, 2023 according to the [Node.js Release Schedule](https://github.com/nodejs/release#release-schedule). Support for a later version of Node.js by GitHub JavaScript actions needs to be made available in good time ahead of this retirement date.

**Comment by MikeMcC399** [comment]
https://github.blog/changelog/2023-05-04-github-actions-all-actions-will-run-on-node16-instead-of-node12/ from May 4, 2023 says "GitHub Actions: All Actions will run on Node16 instead of Node12"

There was no mention of what happens after Node16 though.

**Reply by TomasHubelbauer** [reply]
I would also really like to see an update on this topic. The current stable Node version is Node 20 and Node 16 is going out of support in three months: https://endoflife.date/nodejs

**Comment by TomasHubelbauer** [comment]
Also see (and subscribe to, if interested) the linked issue: https://github.com/actions/runner/issues/2255

**Comment by MikeMcC399** [comment]
This has been provisionally answered in https://github.com/actions/runner/pull/2257#issuecomment-1514803931 by @fhammerl

> With Node16 security updates coming to an end this October, we're of course planning on embedding a new Node.js version, but I can't give an exact date as of today.

**Comment by MikeMcC399** [comment]
- https://github.com/actions/runner/issues/2619 is also related

**Comment by DariuszPorowski** [comment]
@MikeMcC399 @TomasHubelbauer looking for node18 support as well (node20 even better!). Did small PoC with an ugly workaround (wrap js action into composite action): https://github.com/DariuszPorowski/node-js-action
It works, but that may be a kind of a temporary workaround for action maintainers only, not users.
Definitely not a supported solution, and I don't know the potential negative/side effects.

From an action users perspective, the yaml syntax remains the same.

**Reply by andykenward** [reply]
@DariuszPorowski Thanks, this worked for me using node 18.

**Comment by MikeMcC399** [comment]
This has been answered in https://github.com/actions/runner/discussions/2704, where it says "From here, we plan to skip over node 18 and hop on the node 20 LTS release. "

**Comment by MikeMcC399** [comment, marked as answer]
See https://github.com/actions/runner/issues/2619#issuecomment-1679003443

> Hello, latest runner release ([2.308.0](https://github.com/actions/runner/releases/tag/v2.308.0)) contains latest node16 version and node20 version. This runner version will be fully rolled-out by the end of this week. After release you will be able to use `'node20'` actions
